syntax = "proto3";

package inverter;

message InverterStatus {
	uint64 _inner_timestamp = 1;
	uint64 status = 2;
}
message InverterErrors {
	uint64 _inner_timestamp = 1;
	uint64 errors = 2;
}
message InverterTemp {
	uint64 _inner_timestamp = 1;
	double temperature = 2;
}
message InverterMotorTemp {
	uint64 _inner_timestamp = 1;
	double motor_temp = 2;
}
message InverterSpeed {
	uint64 _inner_timestamp = 1;
	double speed = 2;
}
message InverterDasTorque {
	uint64 _inner_timestamp = 1;
	double torque = 2;
}
message InverterInvTorque {
	uint64 _inner_timestamp = 1;
	double torque = 2;
}
message InverterInvTorqueActual {
	uint64 _inner_timestamp = 1;
	double torque = 2;
}

message InverterVec {
	repeated InverterTemp inverter_temp = 1;
	repeated InverterMotorTemp inverter_motor_temp = 2;
	repeated InverterSpeed inverter_speed = 3;
    repeated InverterStatus inverter_status = 4;
    repeated InverterErrors inverter_errors = 5;
	repeated InverterDasTorque inverter_das_torque = 6;
	repeated InverterInvTorque inverter_inv_torque = 7;
	repeated InverterInvTorqueActual inverter_inv_torque_actual = 8;
}

message Inverter {
	InverterVec inverter_l = 1;
	InverterVec inverter_r = 2;
}